daemon/src/rollover_maker.rs

```diff
@@ -0,0 +1,296 @@
+
+        let rollover = setup_contract::roll_over(
+            self.send_to_taker_actor.sink().with(move |msg| {
+                future::ok(maker_inc_connections::TakerMessage {
+                    taker_id,
+                    msg: wire::MakerToTaker::RollOverProtocol { order_id, msg },
+                })
+            }),
+            receiver,
+            (self.oracle_pk, announcement),
+            RolloverParams::new(
+                cfd.order.price,
+                cfd.quantity_usd,
+                cfd.order.leverage,
+                cfd.refund_timelock_in_blocks(),
+                cfd.order.fee_rate,
+            ),
+            Role::Maker,
+            dlc,
+            self.n_payouts,
+        )
+        .await;
+
+        match rollover {
+            Ok(dlc) => self.handle_updated_contract(order_id, dlc, ctx).await?,
+            Err(error) => self.handle_failed(order_id, error, ctx).await?,
+        };
+
+        Ok(())
+    }
+
+    async fn handle_reject_rollover(
+        &mut self,
+        _msg: RejectRollOver,
+        ctx: &mut xtra::Context<Self>,
+    ) -> Result<()> {
+        tracing::info!(%self.cfd.order.id, "Maker rejects a roll_over proposal" );
+
+        self.send_to_taker_actor
+            .send(TakerMessage {
+                taker_id: self.taker_id,
+                msg: MakerToTaker::RejectRollOver(self.cfd.order.id),
+            })
+            .await??;
+        self.projection_actor
+            .send(UpdateRollOverProposal {
+                order: self.cfd.order.id,
+                proposal: None,
+            })
+            .await?;
+        let msg = RollOverCompleted::Rejected {
+            order_id: self.cfd.order.id,
+        };
+        self.maker_cfd_actor.send(msg).await?;
+        ctx.stop();
+        Ok(())
+    }
+
+    async fn handle_updated_contract(
+        &mut self,
+        order_id: OrderId,
+        dlc: Dlc,
+        ctx: &mut xtra::Context<Self>,
+    ) -> Result<()> {
+        let msg = RollOverCompleted::Success { order_id, dlc };
+        self.maker_cfd_actor.send(msg).await?;
+        ctx.stop();
+        Ok(())
+    }
+
+    async fn handle_failed(
+        &mut self,
+        order_id: OrderId,
+        error: Error,
+        ctx: &mut xtra::Context<Self>,
+    ) -> Result<()> {
+        tracing::info!(%self.cfd.order.id, "Rollover failed" );
+        self.projection_actor
+            .send(projection::UpdateRollOverProposal {
+                order: self.cfd.order.id,
+                proposal: None,
+            })
+            .await?;
+        let msg = RollOverCompleted::Failed { order_id, error };
+        self.maker_cfd_actor.send(msg).await?;
+        ctx.stop();
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Handler<AcceptRollOver> for Actor {
+    async fn handle(&mut self, msg: AcceptRollOver, ctx: &mut Context<Self>) {
+        log_error!(self.handle_accept_rollover(msg, ctx));
```

Comment:
We are trying to phase out `log_error`, please don't add any more uses unless really necessary. Have a look at what Lucas did in the other actors for alternatives.